src/background.ts

```diff
@@ -0,0 +1,103 @@
+"use strict";
+
+import { app, protocol, BrowserWindow } from "electron";
+import {
+    createProtocol,
+    // this file is mostly auto generated and imported into another generated project
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    installVueDevtools
+} from "vue-cli-plugin-electron-builder/lib";
+import path from "path";
+
+// hack to get __static working
+// see: https://github.com/electron-userland/electron-webpack/issues/172
+declare const __static: string;
+
+const isDevelopment = process.env.NODE_ENV !== "production";
+
+// Keep a global reference of the window object, if you don't, the window will
+// be closed automatically when the JavaScript object is garbage collected.
+let win: BrowserWindow | null;
+
+// Scheme must be registered before the app is ready
+protocol.registerSchemesAsPrivileged([
+    { scheme: "app", privileges: { secure: true, standard: true } }
+]);
+
+function createWindow(): void {
+    // Create the browser window.
+    win = new BrowserWindow({
+        width: 800,
+        height: 600,
+        icon: path.join(__static, "icon.png"),
+        webPreferences: {
+            nodeIntegration: true
+        }
+    });
+
+    if (process.env.WEBPACK_DEV_SERVER_URL) {
+        // Load the url of the dev server if in development mode
+        win.loadURL(process.env.WEBPACK_DEV_SERVER_URL as string);
+        if (!process.env.IS_TEST) win.webContents.openDevTools();
+    } else {
+        createProtocol("app");
+        // Load the index.html when not in development
+        win.loadURL("app://./index.html");
+    }
+
+    win.on("closed", () => {
+        win = null;
+    });
+}
+
+// Quit when all windows are closed.
+app.on("window-all-closed", () => {
+    // On macOS it is common for applications and their menu bar
+    // to stay active until the user quits explicitly with Cmd + Q
```

Comment:
But do we want this behavior on macOS? I don't think so. We are not a multi-window application.